x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/integration/DlsFlsRequestCacheTests.java

```diff
@@ -342,7 +342,7 @@ public void testRequestCacheWithTemplateRoleQuery() {
         // Since the DLS for the alias uses a stored script, this should cause the request cached to be disabled
         assertSearchResponse(client1.prepareSearch(DLS_TEMPLATE_ROLE_QUERY_ALIAS).setRequestCache(true), Set.of("1"), Set.of("username"));
         // No cache should be used
-        assertCacheState(DLS_TEMPLATE_ROLE_QUERY_INDEX, 2, 2);
+        assertCacheState(DLS_TEMPLATE_ROLE_QUERY_INDEX, 3, 2);
```

Comment:
could you expand on these assertions needing to be adapted?

Reply:
Not quite to be honest. It seems going through the transport layer updates the thread-local context variable somehow and then authz works differently. But I think the security folks need to look at this.
I used the same approach in the batched exec logic (easy to switch to go through the transport layer if need be fortunately). I think this getting a review from security is a soft-dependency to using this path for batched, I'm just not sure how strong tests are end-to-end :)